# Primera entrega
## Curso Data Science 
____________________________________________________

## Presentación de Datasets potenciales
____________________________________________________
### Estudiante: Marvin Cruz Sandoval DNI: 95711072

________________________
_______________________

*Objetivo:* Presentar 3 Datasets propuestos a utilizar durante el curso. 

Se consideraron fuentes de datos relacionadas con el sector energético ya que se busca estudiar la accesibilidad a la energía eléctrica y las perspectivas de incremento de fuentes renovables. 

### 1. Data set de compañías eléctricas a nivel mundial 
Se propone la utilización de este Dataset para analizar las diferentes industrias de  distribución energética a nivel mundial, con el fin de considera el desempeño por region, por pais. En dicho Data set se dispone de una serie de indicadores de manera anualizada. 

Fuente de datos: https://energydata.info/

In [1]:
#Importamos las librerias a utilizar

import numpy as np
import pandas as pd

In [2]:
#Carga de archivo .csv a utilizar

df_1= pd.read_csv("1.global_power_plant_database.csv", sep=";")

#Muestra de encabezado de Data Frame
df_1.head()


country country_long                                              name  \
0     AFG  Afghanistan      Kajaki Hydroelectric Power Plant Afghanistan   
1     AFG  Afghanistan                                      Kandahar DOG   
2     AFG  Afghanistan                                      Kandahar JOL   
3     AFG  Afghanistan     Mahipar Hydroelectric Power Plant Afghanistan   
4     AFG  Afghanistan  Naghlu Dam Hydroelectric Power Plant Afghanistan   

  capacity_mw  latitude longitude primary_fuel commissioning_year  \
0        33.0    323.22   651.190        Hydro                NaN   
1        10.0    316.70   657.950        Solar                NaN   
2        10.0    316.23   657.920        Solar                NaN   
3        66.0    345.56   694.787        Hydro                NaN   
4       100.0    346.41   697.170        Hydro                NaN   

   year_of_capacity_data generation_gwh_2013 generation_gwh_2014  \
0                 2017.0                 NaN                 NaN   
1                    NaN                 NaN                 NaN   
2                    NaN                 NaN                 NaN   
3                 2017.0                 NaN                 NaN   
4                 2017.0                 NaN                 NaN   

  generation_gwh_2015 generation_gwh_2016 generation_gwh_2017  \
0                 NaN                 NaN                 NaN   
1                 NaN                 NaN                 NaN   
2                 NaN                 NaN                 NaN   
3                 NaN                 NaN                 NaN   
4                 NaN                 NaN                 NaN   

  generation_gwh_2018 generation_gwh_2019  
0                 NaN                 NaN  
1                 NaN                 NaN  
2                 NaN                 NaN  
3                 NaN                 NaN  
4                 NaN                 NaN

In [3]:
#Se obtienen las dimensiones del Dataset
df_1.shape

(34936, 16)

Se observa que este Dataset contiene 34936 filas de datos, con 36 columas. 

Respecto a los campos observados, se tienen los siguientes parámetros de relevancia: 

>name:Nombre de la empresa 

>capacity_mw: Capacidad instalada en MW (Mega Watt)

>primary_fuel: Este es un campo de texto, posee información sobre el combustible utilizado por la empresa. 

>year_of_capacity_data: Año en que se reportó el dato de capacidad

### 2. Datos de acceso a electricidad Latinoamérica y el Caribe

Este Dataset contine información sobre los indicadores de acceso a energía eléctrica de los paises de América Latina y el caribe, basado en información proporcionada por diferentes empresas y organismos gubernamentales. 

Fuente de datos: https://energydata.info/

In [4]:
#Carga de archivo .csv a utilizar

df_2= pd.read_csv("2.Latam data energy access_world_bank.csv", sep=";")

#Muestra de encabezado de Data Frame
df_2.head(5)

_id  country                     utility name  utility code  utility ID  \
0    1  country                     utility name  utility code  utility ID   
1    2  country                             name          code    id_conce   
2    3  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.      LOS RIOS      201001   
3    4  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.      LOS RIOS      201001   
4    5  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.      LOS RIOS      201001   

   year   Type of connection  \
0  year   Type of connection   
1  year             typ_conc   
2  1994  Elec - Distribution   
3  1995  Elec - Distribution   
4  1996  Elec - Distribution   

                                                   1  \
0   Total number of connections (residential and ...   
1                                              n_con   
2                                                NaN   
3                                      4.293.683.333   
4                                      4.612.191.667   

                                                   2  \
0  Total Number of residential connections in the...   
1                                           n_rd_con   
2                                                NaN   
3                                      3.694.916.667   
4                                            40018.5   

                                       3  ...  \
0  Total electricity sold per year (MWh)  ...   
1                               mwh_sold  ...   
2                                    NaN  ...   
3                               118861.9  ...   
4                               124118.5  ...   

                                                  16  \
0  Number of residential subscribers per 100 hous...   
1                                           coverage   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                       17  \
0  Number of customer complaints per year   
1                               complains   
2                                     NaN   
3                                     NaN   
4                                     NaN   

                                                  18  \
0  Average duration between customer complaint an...   
1                                        dur_problem   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                                  19  \
0  Average residential tariff (local nominal curr...   
1                                    avg_rd_tariff_p   
2                                                NaN   
3                                      4.985.647.636   
4                                      7.285.148.198   

                                                  20  \
0  Average industrial tariff (local nominal curre...   
1                                    avg_in_tariff_p   
2                                                NaN   
3                                      1.735.030.265   
4                                      1.816.219.628   

                                                  21  \
0  Installation Cost (residential) local nominal ...   
1                                    inst_rd_price_p   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                                  22                     23  \
0  Installation Cost (Industrial) local nominal c...  Year of privatization   
1                                    inst_in_price_p                 d_priv   
2                                                NaN                      0   
3                                                NaN                      0   
4                    

Al obtener las dimensiones mediante el método "shape" se logra determinar que el Data Frame posee 4075 filas y 32 columnas. 

______________________________________________________________________________________________________________________

#### Observación importante: Se verifica que los datos del index se encuentran en la 3a fila del data set (del csv cargado), mientras que la fila 1 contiene un correlativo de las columnas y la fila 2 posee información explicativa del contenido de cada columna. 
  
#### Se genera un array para almacenar la información explicativa del contedido de cada celda y se define la información de la fila No. 3 como encabezados de columna.

In [5]:
df_2_desc = df_2.iloc[0].to_numpy()
df_2_desc

array([1, 'country', 'utility name', 'utility code', 'utility ID', 'year',
       'Type of connection',
       ' Total number of connections (residential and non-residential) in the utility area',
       'Total Number of residential connections in the utility area',
       'Total electricity sold per year (MWh)',
       'Length of distribution network (km) ',
       'Total number of employees ',
       'Average wholesale price paid per MWh (local nominal currency)',
       'Average transmission charges per MWh purchased (local nominal currency)',
       'OPEX (operation expenditures) of the distribution services (local nominal currency)',
       'CAPEX (captial expenditures) of the distribution services (local nominal currency) ',
       'TOTEX (total expenditures) of the distribution network (local nominal currency)',
       'Energy losses in distribution per year (due to technical losses and illegal connections) %',
       'Energy losses in distribution per year due to technical loss

Luego se eliminan las primeras dos filas y se asigna la tercera como "index" para que el Data Frame sea cargado de manera correcta:

In [6]:
df_2.columns = df_2.iloc[1]
df_2=df_2.drop([0,1])
df_2.head()

1  2  country                             name      code id_conce  year  \
2  3  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.  LOS RIOS   201001  1994   
3  4  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.  LOS RIOS   201001  1995   
4  5  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.  LOS RIOS   201001  1996   
5  6  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.  LOS RIOS   201001  1997   
6  7  Ecuador  EMPRESA ELECTRICA LOS RIOS S.A.  LOS RIOS   201001  1998   

1             typ_conc          n_con       n_rd_con  mwh_sold  ... coverage  \
2  Elec - Distribution            NaN            NaN       NaN  ...      NaN   
3  Elec - Distribution  4.293.683.333  3.694.916.667  118861.9  ...      NaN   
4  Elec - Distribution  4.612.191.667        40018.5  124118.5  ...      NaN   
5  Elec - Distribution  5.046.158.333  4.385.358.333  134097.2  ...      NaN   
6  Elec - Distribution  5.229.333.333  4.585.483.333  146304.8  ...      NaN   

1 complains dur_problem avg_rd_tariff_p avg_in_tariff_p inst_rd_price_p  \
2       NaN         NaN             NaN             NaN             NaN   
3       NaN         NaN   4.985.647.636   1.735.030.265             NaN   
4       NaN         NaN   7.285.148.198   1.816.219.628             NaN   
5       NaN         NaN   2.891.591.355     277.406.999             NaN   
6       NaN         NaN   2.985.768.958     295.955.862             NaN   

1 inst_in_price_p d_priv rd_cn_wkr mwh_wkr  
2             NaN      0       NaN     NaN  
3             NaN      0       NaN     NaN  
4             NaN      0       NaN     NaN  
5             NaN      0       NaN     NaN  
6             NaN      0       NaN     NaN  

[5 rows x 32 columns]

Analizando los índices vemos que los parámetros más interesantes a coniderar para el análisis de la accesibilidad a la energía eléctrica son:

>country: Contiene el nombre del pais. 

>typ_conc: Contiene el tipo de conexión manejada por la compañía. 

>whosale_price_p: Indica el precio promedio pagado por MWh (Mega-Watt-hora, unidad de consumo energético) en moneda local. 

>avg_rd_tariff_p: Tarifa promedio nominal residencial. 

>inst_rd_price_p: Costo de instalación residencial nominal.

>inst_rd_price_p: Tarifa promedio nominal industrial. 

>inst_in_price_p: Costo de instalación industrial nominal


### 3. Datos de recursos renovables

Este Dataset contiene información horaria sobre disponibilidad de recurso solar y eolico promedio, de interes para estudiar las variaciones diarias. El periodo comprendido es desde enero hasta octubre de 2023. 

Fuente de datos: https://data.nrel.gov/

In [7]:
#Carga del tercer archivo .csv a utilizar

df_3= pd.read_csv("3.Weather NREL.csv", sep=";")

#Muestra de encabezado de Data Frame
df_3.head()

DATE (MM/DD/YYYY)  HOUR-MST Avg Global Radiation [W/m^2]  \
0          1/1/2023         1                      -0.1120   
1          1/1/2023         2                      -0.2526   
2          1/1/2023         3                      -0.0948   
3          1/1/2023         4                      -0.1798   
4          1/1/2023         5                      -0.1672   

  Avg Zenith Angle [degrees] Avg Azimuth Angle [degrees]  \
0                  1.620.207                     437.710   
1                  1.550.994                     533.509   
2                  1.448.559                     723.599   
3                  1.335.843                     847.451   
4                  1.220.803                     944.756   

   Avg Tower Dry Bulb Temp [deg C]  Avg Deck Dry Bulb Temp [deg C]  \
0                          20.2550                         20.7090   
1                          17.4120                         17.5750   
2                          10.5570                         14.4470   
3                           0.6325                          0.8201   
4                          12.7220                         14.1040   

   Avg Tower Wet Bulb Temp [deg C]  Avg Tower Dew Point Temp [deg C]  \
0                          -0.5593                           -37.891   
1                          -0.8137                           -40.769   
2                         -12.4720                           -42.372   
3                         -17.6550                           -50.672   
4                         -14.0390                           -49.989   

  Avg Tower RH [%] Avg Deck RH [%] Avg Total Cloud Cover [%]  \
0          629.658         623.923                   -10.000   
1          627.450         624.018                   -10.000   
2          650.808         628.723                   -10.000   
3          624.210         614.068                   -10.000   
4          599.523         586.837                   -10.000   

  Avg Opaque Cloud Cover [%]  Avg Avg Wind Speed @ 6ft [m/s]  \
0                    -10.000                          10.766   
1                    -10.000                          11.165   
2                    -10.000                          16.662   
3                    -10.000                          17.031   
4                    -10.000                          12.174   

   Avg Avg Wind Speed @ 33ft [m/s]  Avg Peak Wind Speed @ 6ft [m/s]  \
0                           24.573                           16.958   
1                           21.278                           17.333   
2                           26.020                           21.125   
3                           27.891                           21.500   
4                           20.645                           17.000   

   Avg Peak Wind Speed @ 33ft [m/s]  Avg Precipitation [mm]  Avg Albedo (CM3)  \
0                            32.042                     0.0               0.0   
1                            26.574                     0.0               0.0   
2                            29.984                     0.0               0.0   
3                            31.849                     0.0               0.0   
4                            25.480                     0.0               0.0   

   Avg Temp [deg C]  
0           251.437  
1           251.493  
2           251.487  
3           251.403  
4           251.512

In [8]:
df_3.shape

(6599, 20)

In [9]:
df_3

DATE (MM/DD/YYYY)  HOUR-MST Avg Global Radiation [W/m^2]  \
0             1/1/2023         1                      -0.1120   
1             1/1/2023         2                      -0.2526   
2             1/1/2023         3                      -0.0948   
3             1/1/2023         4                      -0.1798   
4             1/1/2023         5                      -0.1672   
...                ...       ...                          ...   
6594         10/2/2023        19                      -0.0217   
6595         10/2/2023        20                      -0.0707   
6596         10/2/2023        21                       0.2229   
6597         10/2/2023        22                      -0.0565   
6598         10/2/2023        23                       0.2212   

     Avg Zenith Angle [degrees] Avg Azimuth Angle [degrees]  \
0                     1.620.207                     437.710   
1                     1.550.994                     533.509   
2                     1.448.559                     723.599   
3                     1.335.843                     847.451   
4                     1.220.803                     944.756   
...                         ...                         ...   
6594                  1.002.170                   2.735.945   
6595                  1.116.046                   2.837.406   
6596                  1.224.606                   2.953.753   
6597                  1.321.739                   3.097.943   
6598                  1.397.291                   3.284.987   

      Avg Tower Dry Bulb Temp [deg C]  Avg Deck Dry Bulb Temp [deg C]  \
0                             20.2550                         20.7090   
1                             17.4120                         17.5750   
2                             10.5570                         14.4470   
3                              0.6325                          0.8201   
4                             12.7220                         14.1040   
...                               ...                             ...   
6594                         179.3370                        180.6530   
6595                         172.9430                        172.4220   
6596                         144.0470                        143.0380   
6597                         139.7530                        139.4070   
6598                         122.7370                        123.3720   

      Avg Tower Wet Bulb Temp [deg C]  Avg Tower Dew Point Temp [deg C]  \
0                             -0.5593                           -37.891   
1                             -0.8137                           -40.769   
2                            -12.4720                           -42.372   
3                            -17.6550                           -50.672   
4                            -14.0390                           -49.989   
...                               ...                               ...   
6594                          97.7360                            31.658   
6595                          92.4400                            24.609   
6596                          85.8600                            37.476   
6597                          82.9460                            34.459   
6598                         100.0480                            84.190   

     Avg Tower RH [%] Avg Deck RH [%] Avg Total Cloud Cover [%]  \
0             629.658         623.923                   -10.000   
1             627.450         624.018                   -10.000   
2             650.808         628.723                   -10.000   
3             624.210         614.068                   -10.000   
4             599.523         586.837                   -10.000   
...               ...             ...                       ...   
6594          373.073         379.363                   -10.000   
6595          369.523         380.123                   -10.000   
6596          487.547         499.342                   -10.000   
6597          493.2

 Se observa un Data Frame con 6599 filas y 20 columnas. Se determina que los campos más relevantes para un posible estudio son los siguientes: 
 
>DATE (MM/DD/YYYY): Indica la fecha del registro
 
>HOUR-MST: Indica la hora 
 
>Avg Global Radiation [W/m^2]: Radiación solar promedio por metro cuadrado. 
 
>Avg Peak Wind Speed @ 33ft [m/s]: Picos promedio de velocidad de viento, medidos a una altura de 33 pies. Velocidad en m/s.
 
>Avg Precipitation [mm]: Precipitaciones medidas en mm.
 
>Avg Albedo (CM3): Parámetro que define la energía incidente por reflejo del terreno. 
 

## Observaciones generales: 

>En los 3 casos fue requerido el uso del separador ";" debido a error de lectura obtenido inicialmente esto fue corregido agregando el argumento sep=";" en la función *pd.read_csv*.

>Para los casos 1 y 2 se observa una cantidad significativa de valores NAN. 